aiosql/queries.py

```diff
@@ -56,36 +64,29 @@ def fn(self: Queries, conn, *args, **kwargs):
     else:
         raise ValueError(f"Unknown operation_type: {operation_type}")
 
-    fn.__name__ = query_name
-    fn.__doc__ = doc_comments
-    fn.sql = sql  # type: ignore
+    return _query_fn(fn, query_name, doc_comments, sql)
 
-    return fn
 
-
-def _make_async_fn(fn: Callable):
+def _make_async_fn(fn: QueryFn) -> QueryFn:
     async def afn(self: Queries, conn, *args, **kwargs):
         return await fn(self, conn, *args, **kwargs)
 
-    afn.__name__ = fn.__name__
-    afn.__doc__ = fn.__doc__
-    afn.sql = fn.sql  # type: ignore
+    return _query_fn(afn, fn.__name__, fn.__doc__, fn.sql)
+
 
-    return afn
+def _make_ctx_mgr(fn: QueryFn) -> QueryFn:
+    def ctx_mgr(self, conn, *args, **kwargs):
+        return self.driver_adapter.select_cursor(conn, fn.__name__, fn.sql, _params(args, kwargs))
```

Comment:
Beautiful, thanks for catching that this `_params` helper wasn't being used here.